pkg/vault/test/authorizer/TimelockAuthorizerRoot.test.ts

```diff
@@ -0,0 +1,149 @@
+import { advanceTime, currentTimestamp } from '@balancer-labs/v2-helpers/src/time';
+import Vault from '@balancer-labs/v2-helpers/src/models/vault/Vault';
+import { sharedBeforeEach } from '@balancer-labs/v2-common/sharedBeforeEach';
+
+describe('TimelockAuthorizer root', () => {
+  let authorizer: TimelockAuthorizer;
+  let root: SignerWithAddress, nextRoot: SignerWithAddress, user: SignerWithAddress, other: SignerWithAddress;
+
+  before('setup signers', async () => {
+    [, root, nextRoot, user, other] = await ethers.getSigners();
+  });
+
+  sharedBeforeEach('deploy authorizer', async () => {
+    const vault = await Vault.create({
+      admin: root,
+      nextAdmin: nextRoot.address,
+    });
+
+    authorizer = new TimelockAuthorizer(vault.authorizer, root);
+  });
+
+  describe('root', () => {
+    describe('setPendingRoot', () => {
+      let ROOT_CHANGE_DELAY: BigNumberish;
+
+      beforeEach('fetch root change delay', async () => {
+        ROOT_CHANGE_DELAY = await authorizer.instance.getRootTransferDelay();
+      });
+
+      it('sets the nextRoot as the pending root during construction', async () => {
+        expect(await authorizer.instance.getPendingRoot()).to.equal(nextRoot.address);
+      });
+
+      context('when scheduling root change', async () => {
+        function itSetsThePendingRootCorrectly(getNewPendingRoot: () => SignerWithAddress) {
+          it('schedules a root change', async () => {
+            const newPendingRoot: SignerWithAddress = getNewPendingRoot();
+            const expectedData = authorizer.instance.interface.encodeFunctionData('setPendingRoot', [
+              newPendingRoot.address,
+            ]);
+
+            const id = await authorizer.scheduleRootChange(newPendingRoot, [], { from: root });
+
+            const scheduledExecution = await authorizer.getScheduledExecution(id);
+            expect(scheduledExecution.executed).to.be.false;
+            expect(scheduledExecution.data).to.be.equal(expectedData);
+            expect(scheduledExecution.where).to.be.equal(authorizer.address);
+            expect(scheduledExecution.protected).to.be.false;
+            expect(scheduledExecution.executableAt).to.be.at.almostEqual(
+              (await currentTimestamp()).add(ROOT_CHANGE_DELAY)
+            );
+          });
+
+          it('can be executed after the delay', async () => {
+            const newPendingRoot: SignerWithAddress = getNewPendingRoot();
+            const id = await authorizer.scheduleRootChange(newPendingRoot, [], { from: root });
+
+            await expect(authorizer.execute(id)).to.be.revertedWith('ACTION_NOT_YET_EXECUTABLE');
+
+            await advanceTime(ROOT_CHANGE_DELAY);
+            await authorizer.execute(id);
+
+            expect(await authorizer.isRoot(root)).to.be.true;
+            expect(await authorizer.isPendingRoot(newPendingRoot)).to.be.true;
+          });
+
+          it('emits an event', async () => {
```

Comment:
I thought this test would check that `RootChangeScheduled` is correctly emitted.

The contents of this test are correct, but they'd fit better under e.g. 'sets the pending root when executed'